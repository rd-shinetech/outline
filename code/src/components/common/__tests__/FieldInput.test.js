import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FieldInput from '../FieldInput';



let props = undefined;

beforeAll(() => {
    props = {
        input: {},
        type: 'text',
        name: 'category',
        label: 'Category',
        placeholder: 'Category',
        meta: { touched: false, error: {}, warning: {} },
        onChange: jest.fn()
    };

    return props;
});



describe('FieldInput.test.js', () => {
    let wrapper = undefined;

    beforeEach(() => {
        wrapper = shallow(<FieldInput {...props} />);        
        return wrapper;
    });


    it('renders without crashing', () => {
        expect(wrapper.length).toEqual(1);
    });


    it('renders as expected', () => {
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });


});



