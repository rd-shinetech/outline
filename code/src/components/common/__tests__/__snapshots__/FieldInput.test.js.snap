exports[`FieldInput.test.js renders as expected 1`] = `
<div
  className="form-group">
  <label
    htmlFor="category">
    Category
  </label>
  <div
    className="field">
    <input
      className="form-control"
      name="category"
      placeholder="Category"
      type="text" />
  </div>
</div>
`;
